billing/actions/accounts.py

```diff
@@ -329,3 +324,15 @@ def toggle_delinquent_status(account_ids: List[int]):
         When(delinquent=True, then=Value(False)),
         default=Value(True)
     ))
+
+
+def charge_pending_invoices(account_id: str):
+    account = Account.objects.get(id=account_id)
+    pending_invoices = account.invoices.payable().only('pk')
+    logger.info('charge-pending-invoices', pending_invoices=pending_invoices)
+    payments = [
+        invoices.pay_with_account_credit_cards(invoice.pk)
+        for invoice in pending_invoices
+    ]
+    if len(payments) == len(pending_invoices):
+        account.mark_as_compliant()
```

Comment:
Because the management command that marks the accounts as compliant/non-compliant looks at many different aspects of the account, it is possible that it will later mark this exact account as non-compliant.
I don't really know how to fix this (maybe call the code that checks for compliance here?)

Reply:
I think it is difficult because the criteria to run the management command is passed by parameter. So in this point we don't have this criteria to know at which point we consider user as delinquent. However if all pending invoices have a success transaction we know user has no debt so we can mark as compliant, doesn't it?

Reply:
exactly for the analysis.
No for the conclusion.

Reply:
If I'm calling to `pay_with_account_credit_cards` for each pending invoice and in this function calls to `psp.charge_credit_card` with the full amount of the invoice, then why I can't assume after that if the number of pending invoices and the number of success transactions are the same all user debt has been paid?

Reply:
now you are talking about the first remark (the fact that you compare the lengths).
_this_ remark is about how the two concepts of paying invoices and being compliant are mixed up everwhere in the code.
And as you have proven with your demonstration with the runtime parameters: these two concepts are not the same.

Reply:
Answered with the other comment